### DISCUSSION TITLE
Socket.io connection failed error. Need Help BADLY

### DISCUSSION THREAD

**Discussion by hmzakhalid** [socketio/socket.io#4297] [Q&A] [answered]
I've spent hours and hours on this but I can't figure out whats wrong. 
It works fine on Localhost but as soon as I deploy it. It stops working. Following is the full configuration of my client/server/docker
I'm using
- React front end
- Node Backend
- Passport Authentication
- Docker
- Caddy 

#### Socket.io versions
Server `"socket.io": "^4.4.1"`
Client `"socket.io-client": "^4.4.1"`

#### Error on browser console
`WebSocket connection to 'wss://website.tk:3000/ws' failed:`
`connect_error due to server error` <--- from "connect_error" 

#### Server
```js
const app = express();

const server = http.createServer(app);
// Initialize socket.io
const io = new Server(server, {
    cors: {
        origin: "https://website.tk/",
        methods: ["GET", "POST"],
        credentials: true
    }
})

// Cross Origin Resource Sharing (CORS)
app.use(cors({
    origin: 'https://website.tk/',
    credentials: true
}));
.
.
.
server.listen(4000, () => {
    console.log("Server is running on port http://localhost:4000");
});

```

#### Client
```js
const socket = socketIOClient("https://website.tk/", {
      withCredentials: true,
      rejectUnauthorized: false // For testing, Didn't fix it.
    });
    socket.on("connect_error", (err) => {
      console.log(`connect_error due to ${err.message}`);
      console.log(err)
    });
```

#### CaddyFile
```
website.tk {
    reverse_proxy /api/* server:4000
    reverse_proxy client:3000
}
```

If you need more information, just ask.
I'd really appreciate the help.

**Comment by darrachequesne** [comment, marked as answer]
Hi! According to `wss://website.tk:3000/ws`, it seems you are using a custom path (the default path is `/socket.io`). Are you using the same value for the server and the client?

Reference: https://socket.io/docs/v4/client-options/#path

**Reply by hmzakhalid** [reply]
I don't know about the custom path. I've not specified it anywhere. My config for server and client are mentioned above.
As you can see in my caddy file. are Requests from my react app to `/api/*` get rerouted to my server. Would that be the issue?

**Reply by hmzakhalid** [reply]
Hey, So what i did was reroute the default `/socket.io/` path to my server and it worked
```
website.tk {
    reverse_proxy /api/* server:4000
    reverse_proxy /socket.io/* server:4000
    reverse_proxy client:3000
}
```
